requirements.txt

```diff
@@ -1,12 +1,14 @@
 aiofiles>=22.0,<24.0
 aiohttp~=3.0
 altair>=4.2.0,<6.0
+authlib # required for oauth
```

Comment:
Worried about dependency bloat. I wonder if we should maybe not add `authlib` and `itsdangerous` to the dependencies but instead raise an error if someone tries to use the OAuth-related features without them?

Reply:
I understand your concern, that's why I avoided to pin a dependency version. I was hoping it would be ok as they are quite shallow dependencies (`itsdangerous` is pure Python while `Authlib` only has a cryptography dep). 

```
Authlib==1.2.1
  - cryptography [required: >=3.2, installed: 38.0.1]
    - cffi [required: >=1.12, installed: 1.15.1]
      - pycparser [required: Any, installed: 2.21]
itsdangerous==2.1.2
```

Otherwise, would it be ok to define a group like  `gradio[oauth]` for example? And raise an issue to the user if it's not installed. Still not ideal IMO but better than asking to install individual packages.

Adding deps to Spaces would only be part of the solution IMO if we want to allow users to also debug things locally (related to [your comment](https://github.com/gradio-app/gradio/pull/4943#issuecomment-1660850609)).

Reply:
Yes, I like the `gradio[oauth]` + adding the dependencies to Spaces. Should catch most use cases without adding too many dependencies to `gradio`.